extensions/amp-analytics/0.1/transport.js

```diff
@@ -118,13 +124,194 @@ export class Transport {
+          processResponse(msg.data.ampAnalyticsResponse);
+        }
+      });
+    }
+  }
+
+  /**
+   * Called when a creative no longer needs its cross-domain iframe (for
+   * instance, because the creative has been removed from the DOM).
+   * Once all creatives using a frame are done with it, the frame can be
+   * destroyed.
+   * @param {!HTMLDocument} ampDoc The AMP document
+   * @param {!Object<string,string>} transportOptions The 'transport' portion
+   * of the amp-analytics config object
+   */
+  static doneWithCrossDomainIframe(ampDoc, transportOptions) {
+    const frameUrl = Transport.appendHashToUrl_(transportOptions['iframe'],
+        transportOptions['dataHash']);
+    if (Transport.hasCrossDomainFrame(frameUrl)) {
+      if (this.decrementUsageCount_(frameUrl) <= 0) {
+        ampDoc.body.removeChild(Transport.crossDomainFrames[frameUrl].frame);
+        delete Transport.crossDomainFrames[frameUrl];
+      }
+    }
+  }
+
+  /**
+   * Returns whether a url of a cross-domain frame is already known
+   * @param {!string} frameUrl
+   * @return {!boolean}
+   */
+  static hasCrossDomainFrame(frameUrl) {
+    return hasOwn(Transport.crossDomainFrames, frameUrl);
+  }
+
+  /**
+   * Create a cross-domain iframe for third-party vendor anaytlics
+   * @param {!HTMLDocument} ampDoc  The document node of the parent page
+   * @param {!string} frameUrl  The URL, including data hash if
+   * applicable, of the cross-domain iframe
+   * @return {!Element}
+   */
+  static createCrossDomainFrame(ampDoc, frameUrl) {
+    // DO NOT MERGE THIS
+    // Warning: the scriptSrc URL below is only temporary. Don't merge
+    // before getting resolution on that.
+    const frame = createElementWithAttributes(ampDoc, 'iframe', {
+      sandbox: 'allow-scripts',
+      name: JSON.stringify({
+        'scriptSrc': '/examples/analytics-3p-remote-frame-helper.js',
+      }),
+    });
+    loadPromise(frame).then(() => {
+      this.setIsReady_(frameUrl);
+    });
+    setStyles(frame,
+        {width: 0, height: 0, display: 'none',
+         position: 'absolute', top: 0, left: 0});
+    Transport.crossDomainFrames[frameUrl] = {
+      frame,
+      isReady: false,
+      msgQueue: [],
+      usageCount: 1
+    };
+    frame.src = frameUrl; // Intentionally doing this after creating load
+    // promise, rather than in the object supplied to
+    // createElementWithAttribute() above. Want to be absolutely
+    // certain that we don't lose the loaded event.
+    return frame;
+  }
+
+  /**
+   * Indicate that a cross-domain frame is ready to receive messages, and
+   * send all messages that were previously queued for it.
+   * @param {!string} frameUrl The URL (including any data hash) of the frame
+   * @private
+   */
+  static setIsReady_(frameUrl) {
+    const frameData = Transport.crossDomainFrames[frameUrl];
+    frameData.isReady = true;
+    this.sendToCrossDomainIframe_(frameData.frame, frameData.msgQueue);
+    delete frameData.msgQueue;
```

Comment:
Yeah, will need to re-add the queue when throttling is implemented in a future phase. Though there is no need for it in the interim.